File: typescript/test/com/murex/tbw/storage/InMemoryRepository.ts
import {Repository} from "@app/storage/Repository";
import {Invoice} from "@app/purchase";
import * as Immutable from "immutable";

export class InMemoryRepository implements Repository {
  private readonly invoices = new Map<number, Invoice>();

  addInvoice(invoice: Invoice): void {
    this.invoices.set(invoice.id, invoice);
  }

  getInvoiceMap(): Immutable.Map<number, Invoice> {
    return Immutable.Map<number, Invoice>(this.invoices);
  }
}

File: java/src/test/java/org/peaksys/gameoftechs/storage/InMemoryRepository.java
package org.peaksys.gameoftechs.storage;

import org.peaksys.gameoftechs.purchase.Invoice;

import java.util.HashMap;
import java.util.Map;

public class InMemoryRepository implements Repository {
    private final Map<Integer, Invoice> invoiceMap;

    public InMemoryRepository() {
        this.invoiceMap = new HashMap<>();
    }

    @Override
    public void addInvoice(Invoice invoice) {
        invoiceMap.put(invoice.getId(), invoice);
    }

    @Override
    public Map<Integer, Invoice> getInvoiceMap() {
        return invoiceMap;
    }


}
